backend/app/core/main.py

```diff
@@ -0,0 +1,41 @@
+"""
+    app.core.main
+    ~~~~~~~~~~~~~~
+    Defines abstract classes to handle logic and database layer
+"""
+
+from sqlalchemy.orm import Session
+
+class DBSessionContext(object):
+    def __init__(self, db: Session = None):
+        self.db = db
+
+class AppService(DBSessionContext):
+    pass
+
+class AppRepository(DBSessionContext):
+    model = None
+
+    def create(self, data: dict):
+        """
+            Generic method to create a single record
+            depending of the model assigned to the repository
+
+            - Args: data: dict
+
+            - Returns:
+                The created record
+        """
+        object = None
+        try:
```

Comment:
You are doing a few things within create, make a separate function for verification, and for commiting, the failure for verifications gets rolled into any other failures interacting with the db